### PR TITLE
feat: notify organizer if participant declines appointment

### PR DESCRIPTION
**Pull request by JenniferFalkenstein**

### CODE REVIEW

**Comment by JenniferFalkenstein** [comment]
Message Translations:
```
'pupil-decline-appointment-match' --> {"body":"'{{appointment.title}}'","headline":"Terminabsage von {{participant.firstname}}"}

'pupil-decline-appointment-group' --> {"body":"'{{appointment.title}}' {{course.title}}","headline":"Terminabsage von {{participant.firstname}}"}
```

File: graphql/appointment/mutations.ts [hunk 1]
```diff
@@ -161,36 +162,77 @@ export class MutateAppointmentResolver {
...
+        const language = 'de-DE';
+        const appointmentType = appointment.appointmentType;
+        const organizers = await prisma.appointment_organizer.findMany({ where: { appointmentId: appointmentId } });
+        const pupil = await prisma.pupil.findUnique({ where: { id: user.pupilId } });
+
+        if (appointmentType === lecture_appointmenttype_enum.group) {
+            const subCourse = await prisma.subcourse.findFirst({ where: { id: appointment.subcourseId } });
+            const course = await prisma.course.findFirst({ where: { id: subCourse.courseId } });
+            for await (const organizer of organizers) {
+                const student = await getStudent(organizer.studentId);
```
Comment (dhenkel92) [inline graphql/appointment/mutations.ts:207]: The student could be included in the organizer query in line 200, like:

```
        const organizers = await prisma.appointment_organizer.findMany({ where: { appointmentId: appointmentId }, include: { student: true } });
```

File: graphql/appointment/mutations.ts [hunk 2]
```diff
@@ -161,36 +162,77 @@ export class MutateAppointmentResolver {
     @Mutation(() => Boolean)
     @AuthorizedDeferred(Role.ADMIN, Role.OWNER, Role.APPOINTMENT_PARTICIPANT)
     async appointmentDecline(@Ctx() context: GraphQLContext, @Arg('appointmentId') appointmentId: number) {
+        const { user } = context;
+        logger.info(`appointmentId: ${appointmentId}`);
         const appointment = await getLecture(appointmentId);
+        logger.info(`appointment: ${JSON.stringify(appointment)}`);
```
Comment (dhenkel92) [inline graphql/appointment/mutations.ts:168]: Could you please add a proper log message and pass the data as object?
```suggestion
        logger.info(`appointment:`, {appointment: JSON.stringify(appointment, null, 4)});
```

File: graphql/appointment/mutations.ts [hunk 3]
```diff
@@ -161,36 +162,77 @@ export class MutateAppointmentResolver {
...
-        // * Send notification here
+
+        const language = 'de-DE';
+        const appointmentType = appointment.appointmentType;
+        const organizers = await prisma.appointment_organizer.findMany({ where: { appointmentId: appointmentId } });
+        const pupil = await prisma.pupil.findUnique({ where: { id: user.pupilId } });
+
+        if (appointmentType === lecture_appointmenttype_enum.group) {
+            const subCourse = await prisma.subcourse.findFirst({ where: { id: appointment.subcourseId } });
+            const course = await prisma.course.findFirst({ where: { id: subCourse.courseId } });
```
Comment (dhenkel92) [inline graphql/appointment/mutations.ts:205]: This could be one query

File: graphql/appointment/mutations.ts [hunk 4]
```diff
@@ -161,36 +162,77 @@ export class MutateAppointmentResolver {
...
+                        day: appointment.start.toLocaleString(language, { weekday: 'long' }),
+                        date: `${appointment.start.toLocaleString(language, { day: 'numeric', month: 'long', year: 'numeric' })}`,
+                        time: `${appointment.start.toLocaleString(language, { hour: '2-digit', minute: '2-digit' })}`,
+                    },
+                    pupil,
+                    course,
+                });
+            }
+        } else if (appointmentType === lecture_appointmenttype_enum.match) {
+            for await (const organizer of organizers) {
```
Comment (dhenkel92) [inline graphql/appointment/mutations.ts:220]:
```suggestion
            for (const organizer of organizers) {
```

**Pull request merged**